homeassistant/components/tvoverlay/config_flow.py

```
@@ -0,0 +1,79 @@
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TvOverlayFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TvOverlay."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {CONF_HOST: user_input[CONF_HOST], CONF_NAME: user_input[CONF_NAME]}
+            )
+            error, info = await self._async_try_connect(user_input[CONF_HOST])
+            if error is not None:
+                errors["base"] = error
+            elif info is not None:
+                device_name = (
+                    info.get("result", {})
+                    .get("settings", {})
+                    .get("deviceName", DEFAULT_NAME)
+                )
+                device_id = (
+                    info.get("result", {}).get("status", {}).get("id", DEFAULT_NAME)
+                )
+                unique_id = str(device_id).replace("-", "")
+                await self.async_set_unique_id(unique_id)
+                if user_input[CONF_NAME] == DEFAULT_NAME:
+                    user_input[CONF_NAME] = device_name
+                self._abort_if_unique_id_configured()
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data=user_input,
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
```

Don't use a default value, it's misleading to the user. Instead add a `"data_description"` in strings.json where you explain if the user does not enter a name, the name we get from the app will be used.
Moreover, if we get a name from the app, why do we even ask the user for the name here?